### ISSUE TITLE
add search subcommand

### ISSUE THREAD

**Issue by zachvalenta** [pipxproject/pipx#249]
**How would this feature be useful?**

tldr it would be cool if users could search a subset of PyPI packages that were CLIs

```sh
$ pipx search lint

flake8
pylint
```

```sh
$ pipx search dependencies

poetry
pipenv
```

On one hand, pip already offers search functionality. But it could be improved in a couple ways that would help `pipx` as a project:

* surfacing CLIs
* smarter search e.g. when I `pip search postgres`, `pgcli` doesn't come up in the results despite being arguably the best Postgres CLI. I don't know if this is a problem with pip's search, how packages are tagged in PyPI, or lack of tagging from packages maintainers.

**Describe the solution you'd like**

high-level I imagine we could port pip's existing search and utilize PyPI tags to filter down to just CLI pkgs

**Comment by LostAccount** [comment]
This is the one feature I would love, to search. I hope this comment and support for the original comment is well received.

In my view being able to search the pypi index would be most welcome indeed.

Kind regards
— Alex

**Comment by uranusjr** [comment]
I think this ultimately goes to pypa/warehouse#284. tl;dr: PyPI does not offer a way to do what you (and we!) want, and we will need to wait for them to have something before this can ever happen.

I would recommend closing this as a "won't fix" or "upstream" for now, since there is really nothing pipx can do here in the foreseeable future, and this ticket would likely be long forgotten by the time there is something actionable 😞

**Comment by cs01** [comment]
That issue was opened in 2014 and it still hasn’t been fixed 😭. I love the python language, but the ecosystem moves painfully slow. I wish people would just build and ship things, then iterate on them. It will never be perfect the first time anyway.
